### PR TITLE
Cache results of dynamic_binary?

### PR DESCRIPTION
**Pull request by hailstorm350**
This caches the results of MachO.open(...).dylib?

For large projects with hundreds of dependencies, this becomes quite an expensive operation.

### CODE REVIEW

**Comment by hailstorm350** [comment]
Tests forthcoming. Wanted to get proof of concept out here first.

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a CHANGELOG entry to credit yourself! 
You can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

Here's an example of your CHANGELOG entry:

```markdown
* Cache results of dynamic_binary? 
  [hailstorm350](https://github.com/hailstorm350)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: lib/cocoapods/sandbox/file_accessor.rb [hunk 1]
```diff
@@ -366,14 +366,17 @@ def expanded_paths(patterns, options = {})
         result.flatten.compact.uniq
       end
 
+      @@cached_dynamic_binary_results = {}
```
Comment (segiddins) [inline lib/cocoapods/sandbox/file_accessor.rb]: any reason this cache needs to be global and can't be per-instance?

File: CHANGELOG.md [hunk 2]
```diff
@@ -22,6 +22,9 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
 * Show git progress when downloading the CocoaPods Specs repo.  
   [Danielle Tomlinson](https://github.com/dantoml)
   [#5937](https://github.com/CocoaPods/CocoaPods/issues/5937)
+* Cache results of dynamic_binary?
```
Comment (dnkoutso) [inline CHANGELOG.md]: nit extra line here

Comment (dnkoutso) [inline CHANGELOG.md]: Also this is at the wrong section, needs to be on master at the top.

Comment (dnkoutso) [inline CHANGELOG.md]: actually the PR needs a rebase

**Comment by dnkoutso** [comment]
@DanToml any objections on merging this? Speeds up performance, particularly on large projects.

**Comment by DanToml** [comment]
@dnkoutso I'll review properly when I get home but this technically changes behaviour right now.

**Comment by dnkoutso** [comment]
👍

**Comment by dnkoutso** [comment]
I think after the change to a non-static cache this shouldn't affect much imo.

**Pull request merged**